### PR TITLE
Json report provides data of each repetition when test run multiple times

### PR DESCRIPTION
**Pull request by birbalsain**
#### Describe Bug or Feature
When test run multiple times then json report of tests doesn't provide data of every repetition. Only last repetition data are present in it.
In setup of testsuite "TestQstatPerformance" qstat_query_list is appended by each next test . New data are same as previous qstate_query_list data. Because of that unnecessarily test takes more time to check duplicate queries of query_list. Also performance results are getting duplicate measurements. It should be fixed in setup().

#### Describe Your Change
Added each repetition data to the json report with repetition count, total run  time of all repetition.
fixed the queries in setup()

#### Attach Test and Valgrind Logs/Output

Description: Tests from given sources on platforms SUSE15, CENTOS8, CENTOS7
Platforms: SUSE15,CENTOS8,CENTOS7
Profile: regression
|ID|TOTAL|FAIL|ERROR|TIMEDOUT|SKIP|PASS|
|--|--|--|--|--|--|--|
|5032|1064|1|2|0|0|1061|

[json_report_proc.txt](https://github.com/openpbs/openpbs/files/5487775/json_report_proc.txt)
[json_report_config.txt](https://github.com/openpbs/openpbs/files/5487781/json_report_config.txt)

[after_changes_ptl_logs.txt](https://github.com/openpbs/openpbs/files/5267374/after_changes_ptl_logs.txt)
[after_changes_json_report.txt](https://github.com/openpbs/openpbs/files/5267375/after_changes_json_report.txt)
[before_change_json.txt](https://github.com/openpbs/openpbs/files/5182954/before_change_json.txt)

[updated_json_report.txt](https://github.com/openpbs/openpbs/files/5349115/updated_json_report.txt)
[perf_json_report.txt](https://github.com/openpbs/openpbs/files/5368934/perf_json_report.txt)
[non_perf_json.txt](https://github.com/openpbs/openpbs/files/5368936/non_perf_json.txt)

[new_json_report.txt](https://github.com/openpbs/openpbs/files/5375590/new_json_report.txt)

### CODE REVIEW

**Review by visheshh — CHANGES_REQUESTED** [review]
replace 'name' with relevant word for count

**Comment by visheshh** [comment]
@birbalsain Test with other performance tests.

File: test/fw/ptl/utils/plugins/ptl_test_db.py [hunk 1]
```diff
@@ -1612,18 +1613,33 @@ def __init__(self, dbtype, dbpath, dbaccess):
             raise PTLDbError(rc=1, rv=False, msg=_msg)
         elif not self.dbpath.endswith('.json'):
             self.dbpath = self.dbpath.rstrip('.db') + '.json'
+        self._data = {}
         self.jdata = {}
+        self.run_count = 1
```
Comment (hirenvadalia) [inline test/fw/ptl/utils/plugins/ptl_test_db.py:1618]: i suggest to rename this to `cur_repeat_count` or `cur_count` (same in PtlTestTestRunner)

File: test/fw/ptl/utils/plugins/ptl_test_db.py [hunk 2]
```diff
@@ -1612,18 +1613,33 @@ def __init__(self, dbtype, dbpath, dbaccess):
...
 
     def __write_test_data(self, data):
-        prev_data = copy.deepcopy(self.jdata)
-        self.jdata = self.res_data.get_json(data=data, prev_data=prev_data)
-        with open(self.dbpath, 'w') as fd:
-            json.dump(self.jdata, fd, indent=2)
-            fd.write("\n")
+
+        name = "Test_run_count:" + str(self.run_count)
+        if self.run_count != PtlTextTestRunner().run_count:
```
Comment (hirenvadalia) [inline test/fw/ptl/utils/plugins/ptl_test_db.py:1627]: `run_count` in PtlTextTestRunner is class variable, you don't have to instate instance of it to access run_count, you can access it like `PtlTextTestRunner.run_count`

File: test/fw/ptl/utils/plugins/ptl_test_db.py [hunk 3]
```diff
@@ -1612,18 +1613,33 @@ def __init__(self, dbtype, dbpath, dbaccess):
...
+            self.jdata[name] = self._data
+            self._data = self.res_data.get_json(data=data)
+            self.run_count = PtlTextTestRunner().run_count
```
Comment (hirenvadalia) [inline test/fw/ptl/utils/plugins/ptl_test_db.py:1630]: same as above

File: test/fw/ptl/utils/plugins/ptl_test_db.py [hunk 4]
```diff
@@ -1612,18 +1613,33 @@ def __init__(self, dbtype, dbpath, dbaccess):
...
+            name = "Test_run_count:" + str(self.run_count)
```
Comment (hirenvadalia) [inline test/fw/ptl/utils/plugins/ptl_test_db.py:1631]: add space after `:`

File: test/fw/ptl/utils/plugins/ptl_test_db.py [hunk 5]
```diff
@@ -1634,7 +1650,7 @@ def write(self, data, logfile=None):
     def close(self, result=None):
         if result is not None:
             dur = str(result.stop - result.start)
-            self.jdata['test_summary']['test_duration'] = dur
+            self.jdata['tests_duration'] = dur
```
Comment (hirenvadalia) [inline test/fw/ptl/utils/plugins/ptl_test_db.py:1653]: why did you changed this?

**Review by hirenvadalia — CHANGES_REQUESTED** [review]

File: test/fw/ptl/utils/plugins/ptl_test_db.py (hunk 5 above)
Comment (birbalsain) [reply test/fw/ptl/utils/plugins/ptl_test_db.py:1653]: this duration is the total time taken by all repetition of all run tests. So I keep it after all repetition.

Comment (hirenvadalia) [reply test/fw/ptl/utils/plugins/ptl_test_db.py:1653]: Can you please post before and after change json report?

File: test/fw/ptl/utils/plugins/ptl_test_db.py (hunk 4 above)
Comment (birbalsain) [reply test/fw/ptl/utils/plugins/ptl_test_db.py]: done

File: test/fw/ptl/utils/plugins/ptl_test_db.py (hunk 1 above)
Comment (birbalsain) [reply test/fw/ptl/utils/plugins/ptl_test_db.py]: done

**Comment by birbalsain** [comment]
@hirenvadalia @visheshh I have done the requested changes.

**Comment by birbalsain** [comment]
@hirenvadalia @visheshh please have a look.

**Comment by kjakkali** [comment]
1) we don't need test_config in avg_measurements.
2) replace "Tests_run_count: <count>" key with "<count>"

Add json for functional test case.

**Comment by birbalsain** [comment]
@visheshh @hirenvadalia Please have a look on it.

**Comment by visheshh** [comment]
Hi @birbalsain , I observed testconfig for tests are getting cumulative data from previous tests. Please handle that case

**Comment by birbalsain** [comment]
[json_report_config.txt](https://github.com/openpbs/openpbs/files/5487784/json_report_config.txt)
[json_report_proc.txt](https://github.com/openpbs/openpbs/files/5487787/json_report_proc.txt)

**Comment by birbalsain** [comment]
[json_report_config.txt](https://github.com/openpbs/openpbs/files/5491312/json_report_config.txt)
[json_report_proc.txt](https://github.com/openpbs/openpbs/files/5491314/json_report_proc.txt)

**Comment by birbalsain** [comment]
[testparamdecorator_json.txt](https://github.com/openpbs/openpbs/files/5514688/testparamdecorator_json.txt)

**Review by visheshh — APPROVED** [review]
LGTM

**Comment by birbalsain** [comment]
Description: Tests from given sources on platforms SUSE15, CENTOS8, CENTOS7
Platforms: SUSE15,CENTOS8,CENTOS7
Profile: regression
|ID|TOTAL|FAIL|ERROR|TIMEDOUT|SKIP|PASS|
|--|--|--|--|--|--|--|
|5020|1064|0|2|0|0|1062|

**Comment by birbalsain** [comment]
Description: Rerun Only Failed Tests From #5020
Platforms: SUSE15,CENTOS8,CENTOS7
Profile: regression
|ID|TOTAL|FAIL|ERROR|TIMEDOUT|SKIP|PASS|
|--|--|--|--|--|--|--|
|5027|1|0|0|0|0|1|

**Comment by birbalsain** [comment]
Description: Tests from given sources on platforms SUSE15, CENTOS8, CENTOS7
Platforms: SUSE15,CENTOS8,CENTOS7
Profile: regression
|ID|TOTAL|FAIL|ERROR|TIMEDOUT|SKIP|PASS|
|--|--|--|--|--|--|--|
|5032|1064|1|2|0|0|1061|

**Comment by hirenvadalia** [comment]
closing and reopening to restart travis.

**Review by hirenvadalia — APPROVED** [review]

**Pull request merged**